lua/lean/rpc.lua

```diff
@@ -6,7 +6,7 @@ local control = require'plenary.async.control'
 ---@class RpcRef
 
 ---@class Session
----@field bufnr number
+---@field client any vim.lsp.client object
```

Comment:
Forgot to mention this earlier, but neovim [won't send a request unless the client supports it](https://github.com/neovim/neovim/blob/b35de6c525e7c8ad9d64d42f57f348dff608b9e5/runtime/lua/vim/lsp.lua#L1261), so since the requests we're making here seem to be specific enough to Lean 4, it seems to me that this commit and the next (419e8928cb78622480ef207d3bbc027e183e820b) aren't necessary. Did you run into an issue before this commit?

Reply:
As the commit message says:

> Workaround for leanprover/lean4#648

The server supports the notification, but crashes when you send it (after the file is closed).